streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryWindowStore.java

```diff
@@ -462,14 +460,34 @@ public boolean hasNext() {
             }
 
             final Bytes key = getKey(next.key);
-            if (key.compareTo(getKey(keyFrom)) >= 0 && key.compareTo(getKey(keyTo)) <= 0) {
+            if (isKeyWithinRange(key)) {
                 return true;
             } else {
                 next = null;
                 return hasNext();
             }
         }
 
+        private boolean isKeyWithinRange(final Bytes key) {
+            boolean isKeyInRange = false;
+            // split all cases for readability and avoid BooleanExpressionComplexity checkstyle warning
+            if (keyFrom == null && keyTo == null) {
+                // fetch all
+                isKeyInRange = true;
+            } else if (keyFrom == null && key.compareTo(getKey(keyTo)) <= 0) {
+                // start from the beginning
+                isKeyInRange = true;
+            } else if (key.compareTo(getKey(keyFrom)) >= 0 && keyTo == null) {
```

Comment:
nit: let's move `keyTo == null` up first so that if it does not satisfy, we do not need to trigger `getKey` anymore.

Reply:
Think about that a bit more, maybe we can make it simpler as:
```
            if (keyFrom == null && keyTo == null) {
                // fetch all
                return true;
            } else if (keyFrom == null) {
                // start from the beginning
                return key.compareTo(getKey(keyTo)) <= 0;
            } else if (keyTo == null) {
                // end to the last
                return key.compareTo(getKey(keyFrom)) >= 0; 
            } else {
                return key.compareTo(getKey(keyFrom)) >= 0 && key.compareTo(getKey(keyTo)) <= 0;
            }
```

Reply:
Nice refactor! Thanks.